pkgs/tools/X11/xdg-utils/default.nix

```diff
@@ -15,73 +16,233 @@ let
     sha256 = "15gw2nyrqmdsdin8gzxihpn77grhk9l97jp7s7pr7sl4n9ya2rpj";
   };
 
-  perlPath = with perlPackages; makePerlPath [
-    NetDBus XMLTwig XMLParser X11Protocol
+  # Required by the common desktop detection code
+  commonDeps = [ dbus coreutils gnugrep gnused ];
+  commonFakes = [
+    "explorer.exe"
+    "gnome-default-applications-properties"
+    "kde-config"
+    "xprop"
   ];
 
+  # This is still required to work around the eval trickery some scripts do
+  commonPrologue = "${writeText "xdg-utils-prologue" ''
+    export PATH=$PATH:${coreutils}/bin
+  ''}";
+
+  solutions = [
+    {
+      scripts = [ "bin/xdg-desktop-icon" ];
+      interpreter = "${bash}/bin/bash";
+      inputs = commonDeps;
+      fake.external = commonFakes ++ [
+        "gconftool-2"
+        "xdg-user-dir"
+      ];
+      keep."$KDE_SESSION_VERSION" = true;
```

Comment:
I don't know if this env is likely to be runtime fragile or if we can count on it being set to the right thing, but my WIP did have a comment about looking in to the possibility of fixing it at build time.

Reply:
It is currently not, but will be when we ship Plasma 5 and 6 both.